### DISCUSSION TITLE
Is there a list of the human readable names for the "area" field?

### DISCUSSION THREAD

**Discussion by justdave** [weather-gov/api#258] [Q&A] [answered]
Many of the API calls have a field called "area" which is state or marine area codes. The specification only gives a list of valid 2-letter codes for this field, but I can't find anywhere that explains what they stand for. The ones that are states are obvious, but there's a bunch at the end which aren't states (presumably marine zones, etc).

It would be nice if there were an API call that would return a list of the valid area codes and their human readable descriptions (but I guess that's a feature request).

I could just hard-code them for now, if only I knew what they all were. :-)

Thanks!

**Comment by StephenClouse** [comment, marked as answer]
The marine codes are defined in [NWS Directive 10-302](https://www.nws.noaa.gov/directives/sym/pd01003002curr.pdf). We have added these definitions to the OpenAPI documentation in the next release:

- AM: Western North Atlantic Ocean and along U.S. East Coast south of Currituck Beach Light NC following the coastline into Gulf of Mexico to Ocean Reef FL including the Caribbean
- AN: Western North Atlantic Ocean and along U.S. East Coast from Canadian border south to Currituck Beach Light NC
- GM: Gulf of Mexico and along the U.S. Gulf Coast from the Mexican border to Ocean Reef FL
- LC: Lake St. Clair
- LE: Lake Erie
- LH: Lake Huron
- LM: Lake Michigan
- LO: Lake Ontario
- LS: Lake Superior
- PH: Central Pacific Ocean including Hawaiian waters
- PK: North Pacific Ocean near Alaska and along Alaska coastline including the Bering Sea and the Gulf of Alaska
- PM: Western Pacific Ocean including Mariana Island waters
- PS: South Central Pacific Ocean including American Samoa waters
- PZ: Eastern North Pacific Ocean and along U.S. West Coast from Canadian border to Mexican border
- SL: St. Lawrence River above St. Regis
